lib/Bio/EnsEMBL/DataCheck/Checks/CompareSchema.pm

```diff
@@ -181,6 +181,7 @@ sub normalise_table_def {
 
   # Remove keys from table definition since they have been extracted.
   $table =~ s/^((?:PRIMARY |UNIQUE |FOREIGN )*KEY\s.*)\n//gm;
+  $table =~ s/^((?:CONSTRAINT |UNIQUE |KEY ).*)\n//gm;
```

Comment:
May be  the `foreign` should still be kept in case we have a compara MyISAM right? I mean this is somehow the same consequence about the ENGINE tests, am I mistaking?

Reply:
These keys are collected and stored earlier in the script to a hash of keys, they are only removed here for the remainder of the comparison in the `is_deeply` `test`. In this case the above regex was simply missing the removal of the `KEY`s which followed a differing syntax either in the `table.sql` schema or the schema collected directly from the - for example, those that do not end in `KEY` and can start with `CONSTRAINT` - I did try to fix the above regex, but I was worried about breaking it for something I could not test and it seemed the safer option just to include another substitution regex. Is that okay?